ci/prow/make_config.go

```diff
@@ -996,9 +1190,308 @@ func (a *stringArrayFlag) Set(value string) error {
+	var data testGroupTemplateData
+	data.Base.TestGroupName = testGroupName
+	data.GcsLogDir = gcsLogDir
+	data.Extras = extras
+	executeTemplate("test group", testGroupTemplate, data)
+}
+
+// generateDashboard generates the dashboard configuration
+func generateDashboard(repoName string, jobNames []string) {
+	outputConfig(fmt.Sprintf("- name: %s\n  dashboard_tab:", repoName))
+	for _, jobName := range jobNames {
+		testGroupName := getTestGroupName(repoName, jobName)
+		baseOptions := "sort-by-name="
+		extras := make(map[string]string)
+		switch jobName {
+		case "continuous":
+			dashboardTabName := jobName
+			executeDashboardTabTemplate(dashboardTabName, testGroupName, baseOptions, extras)
+
+			if repoName == "knative-serving" {
+				dashboardTabName := "conformance-tests"
+				baseOptions = "include-filter-by-regex=test/conformance\\\\.&sort-by-name="
+				executeDashboardTabTemplate(dashboardTabName, testGroupName, baseOptions, extras)
+			}
+		case "dot-release", "auto-release", "performance", "latency", "api-coverage", "playground":
+			dashboardTabName := jobName
+
+			if jobName == "latency" || jobName == "api-coverage" {
+				baseOptions = "exclude-filter-by-regex=Overall$&group-by-directory=&expand-groups=&sort-by-name="
+			}
+			if jobName == "performance" {
+				baseOptions = "exclude-filter-by-regex=Overall$&group-by-target=&expand-groups=&sort-by-name="
+			}
+			if jobName == "latency" {
+				extras["description"] = "95% latency in ms"
+			}
+			if jobName == "api-coverage" {
+				extras["description"] = "Conformance tests API coverage."
+			}
+
+			executeDashboardTabTemplate(dashboardTabName, testGroupName, baseOptions, extras)
+		case "nightly":
+			dashboardTabName := "release"
+			executeDashboardTabTemplate(dashboardTabName, testGroupName, baseOptions, extras)
+		case "test-coverage":
+			dashboardTabName := "coverage"
+			baseOptions = "exclude-filter-by-regex=Overall$&group-by-directory=&expand-groups=&sort-by-name="
+			executeDashboardTabTemplate(dashboardTabName, testGroupName, baseOptions, extras)
+		default:
+			continue
+		}
+	}
+}
+
+// executeTestGroupTemplate outputs the given dashboard tab config template with the given data
+func executeDashboardTabTemplate(dashboardTabName string, testGroupName string, baseOptions string, extras map[string]string) {
+	var data dashboardTabTemplateData
+	data.Name = dashboardTabName
+	data.Base.TestGroupName = testGroupName
+	data.BaseOptions = baseOptions
+	data.Extras = extras
+	executeTemplate("dashboard tab", dashboardTabTemplate, data)
+}
+
+// getTestGroupName get the testGroupName from the given repoName and jobName
+func getTestGroupName(repoName string, jobName string) string {
+	testGroupName := ""
+	switch jobName {
+	case "continuous", "dot-release", "auto-release", "performance", "latency", "api-coverage", "playground":
+		testGroupName = fmt.Sprintf("ci-%s-%s", repoName, jobName)
```

Comment:
Just return the value, here and everywhere. Then the default case becomes the fallback.